collated/main/A0139772U.md

```diff
@@ -0,0 +1,2562 @@
+
+		assert model != null;	 
+		try {
+			model.deleteTask(taskToDelete);
+			model.getUndoStack().push(this);
+			model.getDeletedStackOfTasks().push(taskToDelete);
+		} catch (TaskNotFoundException pnfe) {
+			assert false : "The target task cannot be missing";
+		}
+		return new CommandResult(String.format(MESSAGE_DELETE_TASK_SUCCESS, taskToDelete));
+	}
+
+```
+###### \java\seedu\whatnow\logic\commands\FindCommand.java
+``` java
+package seedu.whatnow.logic.commands;
+
+import java.util.Set;
+
+/**
+ * Finds and lists all tasks in WhatNow whose name contains any of the argument keywords.
+ * Keyword matching is case sensitive.
+ */
+public class FindCommand extends Command {
+
+    public static final String COMMAND_WORD = "find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all tasks with"
+            + "the specified keywords (case-sensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " Online quiz";
+
+    private final Set<String> keywords;
+
+    public FindCommand(Set<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public CommandResult execute() {
+        return new CommandResult(getMessageForTaskListShownSummary(model.getFilteredTaskList(keywords).size()
+                + model.getFilteredScheduleList(keywords).size()));
+    }
+}
+```
+###### \java\seedu\whatnow\logic\commands\ListCommand.java
+``` java
+package seedu.whatnow.logic.commands;
+
+/**
+ * Lists all tasks in WhatNow to the user.
+ */
+public class ListCommand extends UndoAndRedo{
+
+	public static final String COMMAND_WORD = "list";
+
+	public static final String MESSAGE_SUCCESS = "Listed all tasks";
+
+	public static final String MESSAGE_LIST_NOT_ENTERED = " No previous list command was entered";
+
+	public static final String MESSAGE_LIST_NO_REDO_LIST = " No list command to redo";
+
+	public static final String TASK_STATUS_DONE = "done";
+	
+	public static final String TASK_STATUS_COMPLETED = "completed";
+	
+	public static final String TASK_STATUS_ALL = "all";
+	
+	public static final String TASK_STATUS_INCOMPLETE = "incomplete";
+	
+	public static final String MESSAGE_USAGE = COMMAND_WORD 
+			+ ": the task specified\n"
+			+ "Parameters: done\n"
+			+ "Examples: " + COMMAND_WORD + " done or " + COMMAND_WORD + " all or " + COMMAND_WORD;
+
+	public String type;
+
+	public ListCommand(String type) {
+		this.type = type;
+	}
+
+	private void mapInputToCorrectArgumentForExecution() {
+		if (type.equals(TASK_STATUS_DONE)) {
+			type = TASK_STATUS_COMPLETED;
+		} else if (type.equals(TASK_STATUS_ALL)) {
+			type = TASK_STATUS_ALL;
+		} else {
+			type = TASK_STATUS_INCOMPLETE;
+		}
+	}
+
+	@Override
+	public CommandResult execute() {
+		mapInputToCorrectArgumentForExecution();
+		if (type.equals(TASK_STATUS_ALL)) {
+			model.updateFilteredListToShowAll();
```

Comment:
technically, you can extract these lines to a method and control their behavior by passing appropriate context; that way you can improve SLAP in this method. however, this is not very bad